### PR TITLE
Feature: Closes #4

### LINKED ISSUE #4: Add a script to create all the necessary tables and the db itself
On first start-up, a database should be created in `res/` to provide the discord server with an 'empty' game.

### PR DESCRIPTION
**Pull request by OBlount**
If `res/karp.db` can't be found, the bot can now automatically create the database and tables. That means it creates res/karp.db. Think of it as a 'on-first-startup'.

Also, within `database_maker.py`, there is a class that contains all the SQL to 'create/insert into' tables. This can be easily added to for future tables.

Need your input @Thetvdh to see if there are better ways to do this.

### CODE REVIEW

File: lib/db/database_maker.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import os.path
+import sqlite3
+
+
+class CreateDatabase():
+
+    __PATH = os.path.abspath(f"{os.curdir}/res/karp.db")
+    __DIR = "res"
+
+    # A private static method to check if the db file exists.
```
Comment (Thetvdh) [inline lib/db/database_maker.py:10]: Could add the @staticmethod decorator to make it static potentially. Either that or the @classmethod decorator. Consult with python docs before deciding on this

File: lib/db/database_maker.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    # DOCUMENTATION:
+    # RETURNS:
+    # boolean exists
+    def __check_if_db_exists():
+        return os.path.exists(CreateDatabase.__PATH)
+
+
+    # A static method to create the database file. It will also
```
Comment (Thetvdh) [inline lib/db/database_maker.py:19]: See above comment (Ln10)

**Review by Thetvdh — COMMENTED** [review]
If it all works it looks good. Haven't gone over the refactors and suggested potentially adding some decorators but up to you.

**Comment by OBlount** [comment]
Nice one, thank you. I won't merge just yet though, as I'll probably add the decorators. Think that's a decent idea.

File: lib/db/database_maker.py (hunk 1 above)
Comment (Thetvdh) [reply lib/db/database_maker.py:10]: https://docs.python.org/3/library/functions.html#staticmethod

https://docs.python.org/3/library/functions.html#classmethod

**Pull request merged**